File: backends/java/zappy-application/src/main/java/com/zappymart/application/ports/CustomerRepository.java
package com.zappymart.application.ports;

import com.zappymart.domain.accounts.Customer;
import com.zappymart.domain.shared.EmailAddress;

import java.util.Optional;

public interface CustomerRepository {

    Optional<Customer> byId(String customerId);

    Optional<Customer> byEmail(EmailAddress email);

    Customer save(Customer customer);
}

File: backends/kotlin/zappy-application/src/main/kotlin/nl/zappymart/application/ports/CustomerRepository.kt
package nl.zappymart.application.ports

import nl.zappymart.domain.accounts.Customer
import nl.zappymart.domain.shared.EmailAddress

interface CustomerRepository {

    fun findById(customerId: String): Customer?

    fun findByEmail(email: EmailAddress): Customer?

    fun save(customer: Customer): Customer
}
